### ISSUE TITLE
Automated accessibility (a11y) testing via GitHub CI

### ISSUE THREAD

**Issue by tdonohue** [DSpace/dspace-angular#209]
While Angular itself provides some basic, automated accessibility features (like [ARIA-enabled components](https://angular.io/features)), we should be performing some automated accessibility checks to ensure our new UI meets basic accessibility guidelines.

Ideally, we should be striving for [WCAG 2.0 guidelines](https://www.w3.org/TR/WCAG20/). We should minimally aim for Level A compliance.  But, I'd ask any accessibility experts here to chime in and recommend a compliance level.  https://www.w3.org/WAI/WCAG20/quickref/  (_Note: some of the AA Level requirements are quite content specific, and may be hard to generalize for all DSpace sites_)

Digging around, here's a list of some tools we may be able to use to automate some minimal accessibility testing:

* Pa11y: http://pa11y.org/  (_This seems to be the most promising to me_)
    * Enabling in Travis: http://cruft.io/posts/automated-accessibility-testing-node-travis-ci-pa11y/
* Axe: https://www.axe-core.org/
    * A basic example: https://www.deque.com/blog/accessibility-testing-axe-webdriverjs/
* grunt-accessibility: https://www.npmjs.com/package/grunt-accessibility
* HTML CodeSniffer: https://squizlabs.github.io/HTML_CodeSniffer/
    * Looks to be usable to parse any HTML code for web standards (and used by Pa11y).  Can also be called via CLI using PhantomJS or similar, see README at https://github.com/squizlabs/HTML_CodeSniffer/

I'd encourage others to share tools/ideas/best practices on automating some level of a11y testing (even though I realize a11y testing cannot always be fully automated).  I fully admit I'm not an a11y expert, so suggestions/improvements are welcome!

**Comment by tdonohue** [comment]
We had performed some past Angular accessibility testing (manually) on the original prototype UI, so it may also be of use as a reference: https://github.com/DSpace-Labs/angular2-ui-prototype/issues/113

**Comment by mwoodiupui** [comment]
I've been told that we need to meet WCAG 2.0 AA here, since we're a US public university receiving Federal grants.  What do other jurisdictions require?  Can we pull together a combined profile that will serve all sites?

Yes, there may be aspects of this which are more concerned with the content, for which the best we can do in DSpace code is to stay out of the way.

**Comment by albinoflea** [comment]
We've also adopted WCAG 2.0 AA as a standard, and that seems to be the direction that a lot of higher education sites in the US are headed.

While a lot of the guidelines are going to be content specific, it seems like having mechanisms in place to cover typical use cases would be wise... including appropriate alt text for images is a big issue that we're just starting to grapple with.

**Comment by tdonohue** [comment]
This ticket may need to become higher priority, as we've done a lot of accessibility fixes in 7.0 based on an analysis by Deque.  Now that we've moved to GitHub CI, we need to look more towards GitHub Actions.    We may also want to investigate using [`axe-core/cli`](https://www.npmjs.com/package/@axe-core/cli) and compare with `pa11y-ci` (which the initial PR #356 tried using).  Not yet sure which approach is better.

Here's one possibly useful blog post on automating these sorts of accessibility tests via GitHub Actions: https://www.adrianbolonio.com/en/accessibility-github-actions/

**Comment by tdonohue** [comment]
Update on direction.... Since we switched to Cypress.io, this is much easier to do via `cypress-axe`.  See new PR #1314 which builds on #1301
